aliyun-ecs/src/main/java/org/jclouds/aliyun/ecs/features/SshKeyPairApi.java

```diff
@@ -0,0 +1,127 @@
+@RequestFilters(FormSign.class)
+@QueryParams(keys = { "Version", "Format", "SignatureVersion", "ServiceCode", "SignatureMethod" },
+             values = {"{" + Constants.PROPERTY_API_VERSION + "}", "JSON", "1.0", "ecs", "HMAC-SHA1"})
+public interface SshKeyPairApi {
+
+   @Named("sshKeyPair:list")
+   @GET
+   @QueryParams(keys = "Action", values = "DescribeKeyPairs")
+   @ResponseParser(ParseKeyPairs.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   IterableWithMarker<KeyPair> list(@QueryParam("RegionId") String region, ListKeyPairsOptions options);
+
+   @Named("sshKeyPair:list")
+   @GET
+   @QueryParams(keys = "Action", values = "DescribeKeyPairs")
+   @ResponseParser(ParseKeyPairs.class)
+   @Transform(ParseKeyPairs.ToPagedIterable.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<KeyPair> list(@QueryParam("RegionId") String region);
+
+   @Singleton
+   final class ParseKeyPairs extends ParseJson<KeyPairs> {
+
+      @Inject
+      ParseKeyPairs(final Json json) {
+         super(json, TypeLiteral.get(KeyPairs.class));
+      }
+
+      static class ToPagedIterable extends Arg0ToPagedIterable<KeyPair, ParseKeyPairs.ToPagedIterable> {
+
+         private final ECSComputeServiceApi api;
+
+         @Inject
+         ToPagedIterable(ECSComputeServiceApi api) {
+            this.api = api;
+         }
+
+         @Override
+         protected Function<Object, IterableWithMarker<KeyPair>> markerToNextForArg0(final Optional<Object> arg0) {
+            return new Function<Object, IterableWithMarker<KeyPair>>() {
+               @Override
+               public IterableWithMarker<KeyPair> apply(Object input) {
+                  String regionId = arg0.get().toString();
+                  ListKeyPairsOptions listKeyPairsOptions = ListKeyPairsOptions.Builder.paginationOptions(PaginationOptions.class.cast(input));
+                  return api.sshKeyPairApi().list(regionId, listKeyPairsOptions);
+               }
+            };
+         }
+      }
+   }
+
+   @Named("sshKeyPair:create")
+   @POST
+   @QueryParams(keys = "Action", values = "CreateKeyPair")
+   KeyPairRequest create(@QueryParam("RegionId") String region, @QueryParam("KeyPairName") String keyPairName);
+
+   @Named("sshKeyPair:import")
+   @POST
+   @QueryParams(keys = "Action", values = "ImportKeyPair")
+   KeyPair importKeyPair(@QueryParam("RegionId") String region,
+                         @QueryParam("PublicKeyBody") String publicKeyBody,
+                         @QueryParam("KeyPairName") String keyPairName);
+
+   @Named("sshKeyPair:delete")
+   @POST
+   @QueryParams(keys = "Action", values = "DeleteKeyPairs")
+   Request delete(@QueryParam("RegionId") String region);
+
+   @Named("sshKeyPair:delete")
+   @POST
+   @QueryParams(keys = "Action", values = "DeleteKeyPairs")
+   Request delete(@QueryParam("RegionId") String region, DeleteKeyPairOptions deleteOptions);
+}
```

Comment:
Is there a method to get a single keypair?

Reply:
you mean `delete`? yes one or more. Have changed it as names are not optionals